cli/utilities/utils.py

```diff
@@ -89,3 +94,88 @@ def get_custom_repo_url(base_url, cloud_type="openstack"):
         base_url += "compose/Tools/x86_64/os/"
 
     return base_url
+
+
+def get_nodes_by_ids(nodes, ids):
+    """
+    Fetch nodes using provided substring of nodes
+
+    Args:
+        nodes (list|tuple): CephNode objects
+        ids (list|tuple): node name list (eg., ['node1'])
+
+    Returns:
+        list of nodes identified by given ids
+    """
+    node_details = []
+    for name in ids:
+        node = get_node_by_id(nodes, name)
+        if node:
+            node_details.append(node)
+    return node_details
+
+
+def get_node_by_id(nodes, id):
+    """
+    Fetch node using provided node substring::
+        As per the naming convention used at VM creation, where each node.shortname
+        is framed with hyphen("-") separated string as below, please refer
+        ceph.utils.create_ceph_nodes definition
+            "ceph-<name>-node1-<roles>"
+            name: RHOS-D username or provided --instances-name
+            roles: roles attached to node in inventory file
+        In this method we use hyphen("-") appended to node_name string to try fetch exact node.
+        for example,
+            "node1" ----> "node1-"
+        Note: But however if node_name doesn't follow naming convention as mentioned in
+        inventory, the first searched node will be returned.
+        for example,
+            "node" ----> it might return any node which matched first, like node11.
+        return None, If this cluster has no nodes with this substring.
+
+    Args:
+        nodes: Cli obj containing node details
+        id: node1        # try to use node<Id>
+
+    Returns:
+        node instance (CephVMNode)
+    """
+    for node in nodes:
+        searches = re.findall(rf"{id}?\d*", node.hostname, re.IGNORECASE)
+        for ele in searches:
+            if ele == id:
+                return node
+    return None
+
+
+def build_cmd_from_args(seperator=None, **kw):
+    """This method checks from the dictionary the optional arguments
+    if present it adds them in "cmd" and returns cmd.
+
+    Args:
+        kw (dict) : takes a dictionary as an input.
+
+    Returns:
+        cmd (str): returns a command string.
+
+    eg:
+        Args:
+            seperator: None
+            kw={"uid": "<uid>", "purge-keys": True, "purge-data": True}
```

Comment:
Better to add example with separator as well

Reply:
added